### PR TITLE
First draft of event support for linklater

### PR DESCRIPTION
**Pull request by ggranberry**
Hey, I found myself making a little bot that was triggered off of events and figured you might be open to supporting said behavior in Linklater. I went ahead and sketched up a basic implementation, but it could definitely use a bit of work as my lens skills are beginner level.

### CODE REVIEW

File: lib/Network/Linklater.hs [hunk 1]
```diff
@@ -83,6 +87,16 @@ slash inner req respond = do
     Left msg ->
       respond (responseOf status400 ("linklater: unable to parse request: " <> msg ^. packed))
 
+trigger :: (Event -> Application) -> Application
```
Comment (hlian) [inline lib/Network/Linklater.hs]: I think you meant to type `eventTrigger` here?

**Comment by hlian** [comment]
thanks! looks really good. just that one note

File: lib/Network/Linklater.hs (hunk 1 above)
Comment (ggranberry) [reply lib/Network/Linklater.hs]: Ughhh, that's what happens when I write last second stylistic changes without re-testing.

File: lib/Network/Linklater/Types.hs [hunk 2]
```diff
@@ -122,6 +139,22 @@ commandOfParams params = do
...
 
+eventOfBody :: Data.Aeson.Object -> Either String Event
+eventOfBody obj = do
+  user <- User <$> valueOf "user"
+  channel <- valueOf "channel"
+  eventType <- valueOf "type"
+  case eventType of
+    "link_shared" -> do
+      let links = LinkShared $ obj ^.. ix "links" . values . _Object . ix "url" . _String 
+      Event <$> pure links <*> pure user <*> pure channel
```
Comment (hlian) [inline lib/Network/Linklater/Types.hs]: i think `pure (Event links user channel)` might be easier to read here (same note for a couple lines down)

**Comment by hlian** [comment]
let's merge this sucker!

**Comment by hlian** [comment]
i'll cut a minor point release soon. i guess 4.1 will be the release

**Comment by ggranberry** [comment]
Awesome! It should be pretty easy to add whatever new event types people want to use in the future. I just figured it wasn't really worth trying to implement them all in one go.

**Pull request merged**